### PR TITLE
sd-netlink: tiny fixes

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by yuwata** [comment]
CentOS CI (Arch in KVM) failure is not related. cc @mrc0mmand.
```
archlinux_systemd_ci: [450/2749] Linking target src/udev/fido_id.
    archlinux_systemd_ci: FAILED: src/udev/fido_id 
    archlinux_systemd_ci: cc  -o src/udev/fido_id 'src/udev/3eb56e7@@fido_id@exe/fido_id_fido_id.c.o' 'src/udev/3eb56e7@@fido_id@exe/fido_id_fido_id_desc.c.o' -Wl,--as-needed -Wl,--no-undefined -Wl,-z,relro -Wl,-z,now -fstack-protector -Wl,--gc-sections -Wl,--start-group src/udev/libudev_static.a src/shared/libsystemd-shared-244.so src/udev/libudev-basic.a -Wl,--end-group '-Wl,-rpath,$ORIGIN/:$ORIGIN/../shared:/lib64' -Wl,-rpath-link,/build/build/src/udev -Wl,-rpath-link,/build/build/src/shared -Wl,-rpath-link,/lib64
    archlinux_systemd_ci: /usr/bin/ld: error: /build/build/src/udev/libudev.so.1: file too short
    archlinux_systemd_ci: collect2: error: ld returned 1 exit status
```

File: src/libsystemd/sd-netlink/netlink-message.c [hunk 1]
```diff
@@ -830,7 +830,7 @@ int sd_netlink_message_read_in6_addr(sd_netlink_message *m, unsigned short type,
 static int netlink_container_parse(sd_netlink_message *m,
                                    struct netlink_container *container,
                                    struct rtattr *rta,
-                                   unsigned rt_len) {
+                                   unsigned short rt_len) {
```
Comment (poettering) [inline src/libsystemd/sd-netlink/netlink-message.c]: actually I think we should always use size_t when talking about sizes in memory. If some kernel headers are confused and use something else that's fine, and we should cast it to size_t before using it if necessary. 

Hence, I think this should be changed to size_t here, and instead we should explicitly cast to size_t where the checker complains

File: src/libsystemd/sd-netlink/netlink-message.c [hunk 2]
```diff
@@ -859,9 +859,9 @@ static int netlink_container_parse(sd_netlink_message *m,
 int sd_netlink_message_enter_container(sd_netlink_message *m, unsigned short type_id) {
         const NLType *nl_type;
         const NLTypeSystem *type_system;
+        unsigned short size;
         void *container;
         uint16_t type;
-        size_t size;
```
Comment (poettering) [inline src/libsystemd/sd-netlink/netlink-message.c]: similar here: stick with size_t, only size_t can accurately describe sizes in memory, appropriate for all archs

File: src/libsystemd/sd-netlink/netlink-message.c [hunk 3]
```diff
@@ -953,7 +952,7 @@ int sd_netlink_message_enter_container(sd_netlink_message *m, unsigned short typ
 
 int sd_netlink_message_enter_array(sd_netlink_message *m, unsigned short type_id) {
         void *container;
-        size_t size;
+        unsigned short size;
```
Comment (poettering) [inline src/libsystemd/sd-netlink/netlink-message.c]: ditto

File: src/libsystemd/sd-netlink/netlink-message.c (hunk 1 above)
Comment (yuwata) [reply src/libsystemd/sd-netlink/netlink-message.c]: But the checker is in the macro `RTA_OK()`... So, let's ignore the warning ;-)

**Comment by yuwata** [comment]
Updated. Now `size_t` is used, though this introduce LGTM.com warning. PTAL.

**Comment by poettering** [comment]
But the LGTM thing would be fixed if we cast rt_len to `(unsigned short)` when passing it to RTA_OK(), no? That sounds like the appropriate fix to me, no? i.e. acknowledge the shortcoming of the kernel macro, and cast explicitly just for it, and for nothing else?

**Comment by yuwata** [comment]
OK. Now the size is casted in `RTA_OK()` macro. PTAL.

**Comment by yuwata** [comment]
Hmm... Still LGTM.com dislike this.

**Comment by poettering** [comment]
hmm, maybe trick it out by using a temporary variable? maybe something to report to the lgtm folks as bug?

**Pull request merged**